resources/nersc_supervisor.sh

```diff
@@ -0,0 +1,273 @@
+  declare -i level
+  if [[ "$2" == "" ]]; then return; fi
+  level=$1
+  if [[ ${NS_LOG_DETAIL} -eq 0 ]]; then return; fi
+ 
+  if [[ ${NS_LOG_DETAIL} -ge ${level} ]]
+    then
+      echo "$(date) $2" >> "${NS_LOG}"
+    fi
+  return 0
+ }
+
+###
+# Return info about slurm jobs
+function getrunning {
+  logit 2 "getrunning"
+  if ! rawinfo=$(SQUEUE -u icecubed -q xfer -t 12:00:00 -M escori)
+    then
+      logit 0 "SLURM is not working"
+      return 1
+    fi
+  echo "${rawinfo}"
+  return 0
+}
+
+###
+# Launch a process of the appropriate type
+function launch {
+  if [[ ! -f ${NS_SCRIPT_PATH}/$1".sh" ]]
+    then
+      logit 0 "Submit script $1.sh was not found"
+      return 1
+    fi
+  logit 2 "launch $1"
+  if ! SBATCH  --comments="${1}" -o "${NS_SLURM_LOG}/slurm-$1-%j.out" "${NS_SCRIPT_PATH}/$1.sh"
+    then
+      logit 0 "Submit of $1 failed"
+      return 1
+    fi
+  return 0
+}
+
+###
+# Inspect the slurm logs
+# If completed, move to "seen"
+# If not completed, count and make that available to the main
+#  program for comparison w/ the number of slurm jobs.  We
+#  can have a race condition in which one of the incomplete
+#  slurm jobs completes by the time we ask what's active.
+# I figure the process itself (and its heartbeat) will manage
+#  the error reporting
+function cleanupLogs {
+  declare -i unfinished
+  logit 2 "cleanupLogs"
+  if [[ ! -d ${NS_SLURM_LOG} ]]
+    then
+      logit 0 "Misconfiguration with slurm log ${NS_SLURM_LOG}"
+      return 1
+    fi
+  #
+  fileList=$(/usr/bin/ls "${NS_SLURM_LOG}/"*.out)
+  if [[ "${fileList}" == "" ]]; then return; fi
+  unfinished=0
+  for filename in ${fileList}
+    do
+       stuff=$(/usr/bin/grep -Ei '^START|^END' "${filename}")
+       bc=$(echo "${stuff}" | /usr/bin/awk 'BEGIN{x=0;}{if($1=="START")x=x+1;if($1=="END")x=x+2;}END{print x;}')
+       if [[ "${bc}" == "3" ]]
+         then
+           if ! /usr/bin/mv "${filename}" "${NS_SLURM_LOG_SEEN}/"
+             then return $?; fi
+         else
+           unfinished=$(( unfinished + 1 ))
+         fi
+    done
+  echo "${unfinished}"
+  return 0
+}
+
+###
+# Check the slurm job activity information, count the incomplete entries
+
+function countIncomplete {
+  #
+  logit 2 'Entering countIncomplete'
+  twoweeksbefore=$(/usr/bin/date +%s | /usr/bin/awk '{printf("%s\n",strftime("%Y-%m-%d",$1));}')
+  listing=$(/usr/bin/sacct -n -b -S "${twoweeksbefore}" -M escori -p | /usr/bin/awk '{split($0,b,"|");split(b[1],a,".");if(b[2]!="COMPLETED")print a[1],b[2],b[3],"|";}' |/usr/bin/sort -u)
+  logit 2 "${listing}"
+  echo "${listing}" | /usr/bin/awk '{n=split($0,a,"|");print n-1;}'
+  return 0
+}
+
+
+
+#####
+# More initialization
+# This is where the main work really begins.
+###
+#
+declare -i NS_LOG_DETAIL count expectedcount fc incomplete activeJobs
+
+export NS_DEBUG="True"
+
+###############################
+# Setup for debugging/testing #
+###############################
+
+if [[ ${NS_DEBUG} == "True" ]]
```

Comment:
As this code is re-factored, things like log level and debug flags should be moved to either config or command line (or both).